From f359ea7999899f4cb8601778d919ea02f1577b1d Mon Sep 17 00:00:00 2001
From: Thomas Nizan <tnizan@witekio.com>
Date: Mon, 22 Jun 2020 15:04:29 -0700
Subject: hwmon:(pmbus) Add support Infineon IR38060, IR38062, IR38063

add IR38060, IR38062, IR38063 support in IR38064 driver

diff --git a/Documentation/hwmon/ir38064.rst b/Documentation/hwmon/ir38064.rst
index c455d755a267..aae34bf4e42b 100644
--- a/Documentation/hwmon/ir38064.rst
+++ b/Documentation/hwmon/ir38064.rst
@@ -11,6 +11,30 @@ Supported chips:
     Datasheet: Publicly available at the Infineon webiste
       https://www.infineon.com/dgdl/Infineon-IR38064MTRPBF-DS-v03_07-EN.pdf?fileId=5546d462584d1d4a0158db0d9efb67ca
 
+  * Infineon IR38060
+
+    Prefix: 'ir38060'
+    Addresses scanned: -
+
+    Datasheet: Publicly available at the Infineon webiste
+      https://www.infineon.com/dgdl/Infineon-IR38060M-DS-v03_16-EN.pdf?fileId=5546d4625c167129015c3291ea9a4cee
+
+  * Infineon IR38062
+
+    Prefix: 'ir38062'
+    Addresses scanned: -
+
+    Datasheet: Publicly available at the Infineon webiste
+      https://www.infineon.com/dgdl/Infineon-IR38062M-DS-v03_14-EN.pdf?fileId=5546d462533600a4015355d1717617a8
+
+  * Infineon IR38063
+
+    Prefix: 'ir38063'
+    Addresses scanned: -
+
+    Datasheet: Publicly available at the Infineon webiste
+      https://www.infineon.com/dgdl/Infineon-IR38063M-DS-v03_15-EN.pdf?fileId=5546d4625c167129015c3291f5104cf0
+
 Authors:
       - Maxim Sloyko <maxims@google.com>
       - Patrick Venture <venture@google.com>
diff --git a/drivers/hwmon/pmbus/Kconfig b/drivers/hwmon/pmbus/Kconfig
index 1643382c3c88..400aade4323c 100644
--- a/drivers/hwmon/pmbus/Kconfig
+++ b/drivers/hwmon/pmbus/Kconfig
@@ -68,7 +68,7 @@ config SENSORS_IR38064
 	tristate "Infineon IR38064"
 	help
 	  If you say yes here you get hardware monitoring support for Infineon
-	  IR38064.
+	  IR38060, IR38062, IR38063, and IR38064.
 
 	  This driver can also be built as a module. If so, the module will
 	  be called ir38064.
diff --git a/drivers/hwmon/pmbus/ir38064.c b/drivers/hwmon/pmbus/ir38064.c
index 1820f5077f66..4f06734dd3e3 100644
--- a/drivers/hwmon/pmbus/ir38064.c
+++ b/drivers/hwmon/pmbus/ir38064.c
@@ -42,7 +42,10 @@ static int ir38064_probe(struct i2c_client *client,
 }
 
 static const struct i2c_device_id ir38064_id[] = {
-	{"ir38064", 0},
+	{"ir38060", 30},
+	{"ir38062", 32},
+	{"ir38063", 33},
+	{"ir38064", 34},
 	{}
 };
 
@@ -61,5 +64,5 @@ static struct i2c_driver ir38064_driver = {
 module_i2c_driver(ir38064_driver);
 
 MODULE_AUTHOR("Maxim Sloyko <maxims@google.com>");
-MODULE_DESCRIPTION("PMBus driver for Infineon IR38064");
+MODULE_DESCRIPTION("PMBus driver for Infineon IR38064 and compatible chips");
 MODULE_LICENSE("GPL");
